src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1095,15 +1109,26 @@ private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref
         {
             bool refersToNoPiaLocalType;
 
+            var isConstFound = false;
             SignatureTypeCode typeCode;
-            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
+            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode, acceptRequiredModifier: type =>
+            {
+                var isConst = IsConstModifierType(type);
+                isConstFound |= isConst;
+                return isConst;
+            });
 
             if (typeCode == SignatureTypeCode.ByReference)
             {
                 info.IsByRef = true;
-                info.RefCustomModifiers = info.CustomModifiers; 
+                info.RefCustomModifiers = info.CustomModifiers;
                 info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
             }
+            else if (isConstFound)
```

Comment:
>else if (isConstFound) [](start = 12, length = 22)

We could simply check for presence of any required modifiers in info.CustomModifiers.